tensorflow/python/framework/graph_util_impl.py

```diff
@@ -123,6 +123,9 @@ def extract_sub_graph(graph_def, dest_nodes):
   if not isinstance(graph_def, graph_pb2.GraphDef):
     raise TypeError("graph_def must be a graph_pb2.GraphDef proto.")
 
+  if not isinstance(dest_nodes, list):
```

Comment:
This will break existing code where dest_nodes is a non-list iterable, such as a tuple or set. Instead, you should raise an error if its not an instance of a collections.Iterable, or is an instance of six.string_types as I described above.